src/runtime/wasm.ts

```diff
@@ -291,10 +401,15 @@ export class WasmParticle extends Particle {
       _collectionClear: async (handle) => this.collectionClear(handle),
       _render: (slotName, content) => this.renderImpl(slotName, content),
     };
+
+    const driversToRun = this.wasmDrivers.filter(driver => driver.shouldRun(module));
+    driversToRun.forEach(driver => driver.configureEnvironment(module, this, env));
```

Comment:
We should only ever have one driver for a given module, so I think it would be cleaner to have a factory function that checks for the metadata section once and returns the appropriate driver. Obviously that logic would change as we add new drivers, but that should be trivial.

Reply:
Done.